Doc/reference/simple_stmts.rst

```diff
@@ -860,7 +860,12 @@ the :ref:`relativeimports` section.
 :func:`importlib.import_module` is provided to support applications that
 determine dynamically the modules to be loaded.
 
-.. audit-event:: import module,filename,sys.path,sys.meta_path,sys.path_hooks import
+.. audit-event:: import module_name,filename,sys.path,sys.meta_path,sys.path_hooks import
+.. audit-event:: import.complete module_name,module import
+
+The ``import`` audit event may be raised twice, when an extension module
+is imported. In all cases ``import.complete`` audit event is matched by an
+``import`` audit event with ``filename=None``.
```

Comment:
Let's phrase the first part as "When importing an extension module, the ``import`` event may be raised twice" ("will"? Why only sometimes?)

I don't understand what you're trying to suggest by the second part... is it suggesting that an ``import/module`` (after rename) event will always be preceded by ``import``? Perhaps it'll make more sense to explain when you could get ``import`` _without_ ``import/module``? (e.g. on error, module was cached, etc.).

Reply:
It will always be preceded by one import event, but it may be preceded by two, and anything that is using this mechanism to profile imports needs to know that.

Currently there are two places that an `import` audit event may be generated.  The first one - in `import_find_and_load` - occurs for all imports. The second - in `_imp_create_dynamic_impl` - occurs only under some circumstances. This is at least when loading an extension module, but might also be when dynamically creating a module? It's a bit unfortunate that the latter has the same name, but I doubt that can be changed now.

This means that profiling imports requires either a balancing `import/module` for the second case, or being able to differentiate the first and second `import` cases (which can currently be done based upon the second audit event argument). I opted for the latter, because that matches the existing behaviour of PYTHONPROFILEIMPORTTIME and the dtrace events.

Reply:
Yeah, we shouldn't require balancing. We've got no chance of getting that perfect, and _that_ will upset people, so let's just be a bit unpredictable.

Generally we only raise "result" events when the result is going to appear in a later event, so there's some way to correlate those (e.g. `socket.__new__`), or if there may be some reason to abort the operation based on the result (_very_ uncommon). I'm not sure either of those apply here, though probably the latter if someone wants to do deeper analysis of a module _after_ it has run arbitrary code (i.e. this is not a security boundary).

So perhaps what we want to say here is that one or more `import` events may occur for the same module name, and a successful first import of a module will be followed by an `import/module` event?

Reply:
WDYT of the rewording? I did mention that `import/module` also happens if loading fails (verified in the test I added).